staging/src/k8s.io/apiserver/pkg/admission/plugin/initialization/initialization.go

```diff
@@ -183,8 +188,13 @@ func (i *initializer) Admit(a admission.Attributes) (err error) {
 			// Mirror pods are exempt from initialization because they are created and initialized
 			// on the Kubelet before they appear in the API.
 			// TODO: once this moves to REST storage layer, this becomes a pod specific concern
-			if pod, ok := a.GetObject().(*api.Pod); ok && pod != nil {
-				if _, isMirror := pod.Annotations[api.MirrorPodAnnotationKey]; isMirror {
+			if a.GetKind().GroupKind() == v1.SchemeGroupVersion.WithKind("Pod").GroupKind() {
```

Comment:
TODO: move this logic out of k8s.io/apiserver

Reply:
@p0lyn0mial do you have thoughts on how to do this? Mabye moving this logic to a separate kube admission plugin?

Reply:
nope, I haven't thought much about `initiailzers`